pkg/neorpc/filters.go

```diff
@@ -126,14 +146,65 @@ func (f *NotificationFilter) Copy() *NotificationFilter {
 		res.Name = new(string)
 		*res.Name = *f.Name
 	}
+	if len(f.Parameters) != 0 {
+		res.Parameters = slices.Clone(f.Parameters)
+	}
+	f.cachedSI = nil
 	return res
 }
 
+// ParametersSI returns [stackitem.Item] version of [NotificationFilter.Parameters]
+// according to [smartcontract.Parameter.ToStackItem]; result is cached.
+// It mainly should be used by server code. Must not be used concurrently.
+func (f *NotificationFilter) ParametersSI() ([]stackitem.Item, error) {
+	if f.cachedSI != nil {
+		return f.cachedSI, nil
+	}
+	f.cachedSI = make([]stackitem.Item, 0, len(f.Parameters))
+	for i, p := range f.Parameters {
+		si, err := p.ToStackItem()
+		if err != nil {
+			f.cachedSI = nil
+			return nil, fmt.Errorf("converting %d parameter it stack item: %w", i, err)
```

Comment:
if it was possible to escape this check, i would prefer to do it. but since we do it anyway and get `error`, i'd like to keep it. who knows how this structure will be used over time. getter is getter, and the returned value must always be correct, IMO. without this check, it can return unexpected things. but i agree that it is the only reason `error` is returned from `ParametersSI` so ping me one more time if i was not convincing enough, i will change

Reply:
OK, it's not expensive to keep it.